h/services/annotation_read.py

```diff
@@ -0,0 +1,104 @@
+from typing import Iterable, List, Optional
+
+from sqlalchemy import select
+from sqlalchemy.orm import Query, Session, subqueryload
+
+from h import i18n
+from h.db.types import InvalidUUID
+from h.models import Annotation, DocumentURI
+from h.util.uri import normalize
+
+_ = i18n.TranslationStringFactory(__package__)
+
+
+class AnnotationReadService:
+    """A service for storing and retrieving annotations."""
+
+    def __init__(self, db_session: Session):
+        self._db = db_session
+
+    def get_annotation_by_id(self, id_: str) -> Optional[Annotation]:
+        """
+        Fetch the annotation with the given id.
+
+        :param id_: Annotation ID to retrieve
+        """
+        try:
+            return self._db.query(Annotation).get(id_)
+        except InvalidUUID:
+            return None
+
+    def get_annotations_by_id(
+        self, ids: List[str], eager_load: Optional[List] = None
+    ) -> Iterable[Annotation]:
+        """
+        Get annotations in the same order as the provided ids.
+
+        :param ids: the list of annotation ids
+        :param eager_load: A list of annotation relationships to eager load
+            like `Annotation.document`
+        """
+
+        if not ids:
+            return []
+
+        annotations = self._db.execute(
+            self._annotation_search_query(ids=ids, eager_load=eager_load)
+        ).scalars()
+
+        return sorted(annotations, key=lambda annotation: ids.index(annotation.id))
+
+    def search_annotations(
+        self,
+        ids: Optional[List[str]] = None,
+        target_uri: Optional[str] = None,
+        document_uri: Optional[str] = None,
+    ) -> Iterable[Annotation]:
```

Comment:
I wonder if this is more of a "get" method than a "search" method? I think it just gets all the annotations with the given `target_uri` and/or `document_uri` much like the two `get_*()` methods above get annotations with the given id(s), there's no actual "searching" as-such going on. And it prevents overloading the term "search" (when there's presumably another method elsewhere that actually searches annotation contents using Elasticsearch).

Also wondering if the three methods could actually just be one `get()` method with lots of optional arguments.